conans/client/generators/premake_multi.py

```diff
@@ -0,0 +1,86 @@
+        self.defines = ", ".join('"%s"' % p for p in deps_cpp_info.defines)
+        self.cxxflags = ", ".join('"%s"' % p for p in deps_cpp_info.cxxflags)
+        self.cflags = ", ".join('"%s"' % p for p in deps_cpp_info.cflags)
+        self.sharedlinkflags = ", ".join('"%s"' % p.replace('"', '\\"') for p in deps_cpp_info.sharedlinkflags)
+        self.exelinkflags = ", ".join('"%s"' % p.replace('"', '\\"') for p in deps_cpp_info.exelinkflags)
+
+        self.rootpath = "%s" % deps_cpp_info.rootpath.replace("\\", "/")
+
+
+class premake_multi(Generator):
+    @property
+    def filename(self):
+        return "conanbuildinfo_{0}.premake.lua".format(str(self.settings.get_safe("build_type"))).lower()
+
+    @property
+    def content(self):
+        deps = PremakeDeps(self.deps_build_info)
+        lua_module = "conan_{0}".format(str(self.settings.get_safe("build_type"))).lower()
+
+        template = ('{lua_module}.conan_includedirs{dep} = {{{deps.include_paths}}}\n'
+                    '{lua_module}.conan_libdirs{dep} = {{{deps.lib_paths}}}\n'
+                    '{lua_module}.conan_bindirs{dep} = {{{deps.bin_paths}}}\n'
+                    '{lua_module}.conan_libs{dep} = {{{deps.libs}}}\n'
+                    '{lua_module}.conan_system_libs{dep} = {{{deps.system_libs}}}\n'
+                    '{lua_module}.conan_defines{dep} = {{{deps.defines}}}\n'
+                    '{lua_module}.conan_cxxflags{dep} = {{{deps.cxxflags}}}\n'
+                    '{lua_module}.conan_cflags{dep} = {{{deps.cflags}}}\n'
+                    '{lua_module}.conan_sharedlinkflags{dep} = {{{deps.sharedlinkflags}}}\n'
+                    '{lua_module}.conan_exelinkflags{dep} = {{{deps.exelinkflags}}}\n')
+
+        sections = ["#!lua"]
+
+        sections.extend(
+                [
+                 'local {0} = {{}}'.format(lua_module),
+                 "",
+                 '{lua_module}.conan_build_type = "{0}"'.format(str(self.settings.get_safe("build_type")), lua_module=lua_module),
+                 '{lua_module}.conan_arch = "{0}"'.format(str(self.settings.get_safe("arch")), lua_module=lua_module),
+                 ""]
+        )
+
+        all_flags = template.format(lua_module=lua_module, dep="", deps=deps)
+        sections.append(all_flags)
+        template_deps = template + '{lua_module}.conan_rootpath{dep} = "{deps.rootpath}"\n'
+
+        for dep_name, dep_cpp_info in self.deps_build_info.dependencies:
+            deps = PremakeDeps(dep_cpp_info)
+            dep_name = dep_name.replace("-", "_")
+            dep_flags = template_deps.format(lua_module=lua_module, dep="_" + dep_name, deps=deps)
+            sections.append(dep_flags)
+
+        sections.append(
+            "function {lua_module}.conan_basic_setup()".format(lua_module=lua_module)
+        )
+        sections.append(
+            f"    configurations{{{lua_module}.conan_build_type}}\n"
```

Comment:
Do not use f-strings yet, we need to pass CI

Reply:
Hm, the CI run passed with it. Is there another, more thorough CI? Anyway, I replaced them.